### PR TITLE
Add CI workflow using Github Actions

### PR DESCRIPTION
**Pull request by sujithvm**
Integrating CI workflow with Github actions.
This workflow will be triggered for any pull request or push to master branch.
Workflow will run only after it is merged to master.

Tasks which will be implemented in future PRs
- Checkstyle enforcement
Tasks not implemented yet due to approvals
- Codecov code coverage with Jacoco

Testing:
https://github.com/sujithvm/security-advanced-modules/commit/8a75ace9bc82bdc10bb08dc68e7e1c1bf450cd59/checks?check_suite_id=354327804

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        repository: opendistro-for-elasticsearch/security
+        ref: refs/heads/master
+    - name: Install security
+      run: mvn clean install -DskipTests --file ../security/pom.xml
+
+    - name: Checkout security-advanced-modules
+      uses: actions/checkout@v1
+
+    - name: Checkstyle
+      run: mvn checkstyle:checkstyle
```
Comment (allenyin96) [inline .github/workflows/ci.yml:38]: I thought we are skipping this?

Comment (sujithvm) [reply .github/workflows/ci.yml:38]: `mvn checkstyle:checkstyle` will report all the errors using default checkstyle but won't make the build job fail. Once we write our own checkstyle rules and use maven checkstyle plugin, we will run `mvn checkstyle:check` to make the job fail if it violates the standards.

Comment (sujithvm) [reply .github/workflows/ci.yml:38]: From github actions run: `[INFO] There are 5834 errors reported by Checkstyle 8.19 with sun_checks.xml ruleset.`

Comment (allenyin96) [reply .github/workflows/ci.yml:38]: Got it.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    runs-on: ubuntu-latest
+    container: opendistroforelasticsearch/security-maven:v1
+
+    steps:
+
+    - name: Checkout security-parent
+      uses: actions/checkout@v1
+      with:
+        repository: opendistro-for-elasticsearch/security-parent
+        ref: refs/heads/master
```
Comment (allenyin96) [inline .github/workflows/ci.yml:22]: Is this always checking out master? Suppose someone raises a PR against security opendistro-1.1 branch and security-parent master has the code for opendistro-1.3, how will the action work in this case?

Comment (sujithvm) [reply .github/workflows/ci.yml:22]: The user must modify this workflow file in the PR for testing. Point to refs/head/branchname or tag or commit

Comment (allenyin96) [reply .github/workflows/ci.yml:22]: I see. Then we should also update the Readme or another appropriate place to reflect this.

**Review by allenyin96 — APPROVED** [review]

**Review by allenyin96 — APPROVED** [review]

**Review by alolita — APPROVED** [review]
Lgtm.

**Pull request merged**